src/Model/User.php

```diff
@@ -840,6 +840,57 @@ public static function isNicknameBlocked($nickname)
 		return false;
 	}
 
+	/**
+	 * Get avatar link for given user id
+	 *
+	 * @param integer $uid     user id
+	 * @param string  $size    One of the ProxyUtils::SIZE_* constants
+	 * @return string avatar link
+	 */
+	public static function getAvatarUrlForId(int $uid, string $size = ''):string
+	{
+		$url = DI::baseUrl() . '/photo/';
+
+		switch ($size) {
+			case Proxy::SIZE_MICRO:
+				$url .= 'micro/';
+				$scale = 6;
+				break;
+			case Proxy::SIZE_THUMB:
+				$url .= 'avatar/';
+				$scale = 5;
+				break;
+			default:
+				$url .= 'profile/';
+				$scale = 4;
+				break;
+		}
+
+		$updated =  '';
+		$imagetype = IMAGETYPE_JPEG;
+
+		$photo = Photo::selectFirst(['type', 'created', 'edited', 'updated'], ["scale" => $scale, 'uid' => $uid, 'profile' => true]);
+		if (!empty($photo)) {
+			$updated = max($photo['created'], $photo['edited'], $photo['updated']);
+
+			switch ($photo['type']) {
+				case 'image/png':
+					$imagetype = IMAGETYPE_PNG;
+					break;
+
+				case 'image/gif':
+					$imagetype = IMAGETYPE_PNG;
+					break;
+
+				default:
+					$imagetype = IMAGETYPE_JPEG;
+					break;
+			}
+		}
+
+		return $url . $uid . image_type_to_extension($imagetype) . ($updated ? '?ts=' . strtotime($updated) : '');
```

Comment:
We shouldn't publish URLs using any of our auto-incremented IDs as it makes them easily enumerable. In this case, we could be using the nickname since it's unique and already formatted for URLs.

Reply:
We are using this scheme for ages, I haven't introduced it here. Moving to something different is surely a good idea, but in this late stage of the RC I hadn't wanted to change too much.

Reply:
I get it, at least we can change the URL pattern easily now!

Reply:
Yes, this is one of the benefits of this change. We now have two central functions for either the contacts or the users. So we can change it easily from up here.